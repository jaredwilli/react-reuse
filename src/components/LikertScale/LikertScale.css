.likert-scale {
    display: inline-block;
    position: relative;
    height: 80px;
  }

  .likert-scale .label {
    font-size: 14px;
    font-family: BrownStd, sans-serif, Arial;
    text-transform: lowercase;
    margin-bottom: 8px;
    position: relative;
    left: -5px;
  }

  .likert-scale .label::first-letter {
    text-transform: capitalize;
  }

  .likert-scale .line {
    background: #959595;
    position: relative;
    top: 6px;
    height: 1px;
    z-index: 0;
  }

  .likert-scale .dot {
    background: #29292a;
    border-radius: 100%;
    display: inline-block;
    position: absolute;
    top: 25.66px;
    width: 10px;
    height: 10px;
    z-index: 1;
  }

  .likert-scale .dot-points {
    position: relative;
    left: -3px;
    top: 4px;
  }

  .likert-scale .small-dot {
    background: #959595;
    border-radius: 100%;
    width: 5px;
    height: 5px;
    position: absolute;
  }

  .likert-scale .small-dot span {
    color: #797979;
    font-size: 12px;
    font-family: BrownStd, sans-serif, Arial;
    position: absolute;
    top: 15px;
    left: -25px;
    width: 75px;
  }

  .likert-scale .small-dot:first-child span {
    left: 0;
  }
  .likert-scale .small-dot:last-child span {
    text-align: right;
    right: 0;
    left: initial;
  }
