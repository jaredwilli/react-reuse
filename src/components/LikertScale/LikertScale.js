import React from 'react';
import PropTypes from 'prop-types';

import './LikertScale.css';

const LikertScale = props => {
    const valueIndex = props.options.indexOf(props.value.label[0]);
    const valueArr = new Array(props.options.length);

    const labelArrs = {
        size: [
            'Runs Small',
            'True to Size',
            'Runs Large'
        ],
        width: [
            'Runs Narrow',
            'True to Width',
            'Runs Wide'
        ]
    };

    for (let i = 0; i < valueArr.length; i++) {
        if (i === 0) {
            valueArr[i] = 0;
        } else if (i === valueArr.length) {
            valueArr[i] = 100;
        } else {
            valueArr[i] = 100 / (valueArr.length - 1) * i;
        }
    }

    // Calculate the left position for the big black dot
    const dotStyle = {
        left: `${valueArr[valueIndex] - 2.5}%`
    };

    return (
        <div className="likert-scale" style={{ width: props.width }}>
            <div className="label">{props.value.label}</div>

            <div className="dot-points">
                {valueArr.map((val, index) => {
                    const label = labelArrs[props.value.dimensionLabel.toLowerCase()];

                    return (
                        <div key={index} className="small-dot" style={{ left: `${val}%` }}>
                            {([0,2,4].indexOf(index) !== -1) &&
                                <span>
                                    {label[[0,2,4].indexOf(index)]}
                                </span>
                            }
                        </div>
                    );
                })}
            </div>

            <div className="dot" style={dotStyle}></div>
            <div className="line" style={{ width: props.width }}></div>
        </div>
    );
};

LikertScale.propTypes = {
    /** The value of the chart where the dot should display */
    value: PropTypes.object.isRequired,

    /** The list of scale options for setting the values of the dot points */
    options: PropTypes.array.isRequired,

    /** The list of scale options for setting the values of the dot points */
    width: PropTypes.string
};

LikertScale.defaultProps = {
    width: '100%'
};

export default LikertScale;
