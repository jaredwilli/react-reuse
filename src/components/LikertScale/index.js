import LikertScale from './LikertScale';

export default LikertScale;
