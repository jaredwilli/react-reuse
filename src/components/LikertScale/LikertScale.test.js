import _ from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LikertScale from './LikertScale';

describe('LikertScale', () => {
    const values = [
        {
            type: 3,
            dimensionId: 3,
            dimensionLabel: 'Size',
            label: ['Felt True to Size']
        },
        {
            type: 3,
            dimensionId: 3,
            dimensionLabel: 'Size',
            label: ['Felt Full Size Smaller']
        },
        {
            type: 3,
            dimensionId: 3,
            dimensionLabel: 'Size',
            label: ['Felt Half Size Larger']
        },
        {
            type: 3,
            dimensionId: 2,
            dimensionLabel: 'Width',
            label: ['Felt Slightly Narrow']
        },
        {
            type: 3,
            dimensionId: 2,
            dimensionLabel: 'Width',
            label: ['Felt Wide']
        }
    ];

    const SIZE_OPTIONS = [
        'Felt Full Size Smaller',
        'Felt Half Size Smaller',
        'Felt True to Size',
        'Felt Half Size Larger',
        'Felt Full Size Larger'
    ];

    const WIDTH_OPTIONS = [
        'Felt Narrow',
        'Felt Slightly Narrow',
        'Felt True to Width',
        'Felt Slightly Wide',
        'Felt Wide'
    ];

    it('renders the component correctly', () => {
        expect(renderer.create(
            <LikertScale value={values[0]} options={SIZE_OPTIONS} />
        ).toJSON()).toMatchSnapshot();

        expect(renderer.create(
            <LikertScale value={values[1]} options={SIZE_OPTIONS} />
        ).toJSON()).toMatchSnapshot();

        expect(renderer.create(
            <LikertScale value={values[2]} options={SIZE_OPTIONS} />
        ).toJSON()).toMatchSnapshot();

        expect(renderer.create(
            <LikertScale value={values[3]} options={WIDTH_OPTIONS} />
        ).toJSON()).toMatchSnapshot();

        expect(renderer.create(
            <LikertScale value={values[4]} options={WIDTH_OPTIONS} />
        ).toJSON()).toMatchSnapshot();
    });
});
