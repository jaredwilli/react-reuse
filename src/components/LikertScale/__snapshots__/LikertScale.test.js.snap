// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikertScale renders the component correctly 1`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Small
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Size
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Large
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "47.142857142857146%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly 2`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Small
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Size
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Large
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "-2.857142857142857%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly 3`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Small
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Size
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Large
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "72.14285714285714%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly 4`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Narrow
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Width
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Wide
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "22.142857142857142%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly 5`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Narrow
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Width
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Wide
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "97.14285714285714%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly for size 1`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Small
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Size
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Large
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "47.142857142857146%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly for size 2`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Small
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Size
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Large
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "-2.857142857142857%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly for size 3`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Small
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Size
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Large
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "72.14285714285714%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly for size 4`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Narrow
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Width
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Wide
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "22.142857142857142%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;

exports[`LikertScale renders the component correctly for size 5`] = `
<div
  className="likert-scale"
>
  <div
    className="dot-points"
  >
    <div
      className="small-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <span>
        Runs Narrow
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "25%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <span>
        True to Width
      </span>
    </div>
    <div
      className="small-dot"
      style={
        Object {
          "left": "75%",
        }
      }
    />
    <div
      className="small-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <span>
        Runs Wide
      </span>
    </div>
  </div>
  <div
    className="dot"
    style={
      Object {
        "left": "97.14285714285714%",
      }
    }
  />
  <div
    className="line"
    style={
      Object {
        "width": 300,
      }
    }
  />
</div>
`;
